lib/sql.js

```diff
@@ -156,200 +228,892 @@ SqlConnector.prototype.tableEscaped = function (model) {
+ * @returns {string} The list of fields for update
+ */
+SqlConnector.prototype.buildFieldsForUpdate = function(model, data, excludeIds) {
+  if (excludeIds === undefined) {
+    excludeIds = true;
+  }
+  var fields = this.buildFields(model, data, excludeIds);
+  var columns = new ParameterizedSQL('');
+  for (var i = 0, n = fields.names.length; i < n; i++) {
+    var clause = ParameterizedSQL.append(fields.names[i],
+      fields.columnValues[i], '=');
+    columns.merge(clause, ',');
+  }
+  columns.sql = 'SET ' + columns.sql;
+  return columns;
+};
+
+/**
+ * Build a list of escaped column names for the given model and fields filter
+ * @param {string} model Model name
+ * @param {object} filter The filter object
+ * @returns {string} Comma separated string of escaped column names
+ */
+SqlConnector.prototype.buildColumnNames = function(model, filter) {
+  var fieldsFilter = filter && filter.fields;
+  var cols = this.getModelDefinition(model).properties;
+  if (!cols) {
+    return '*';
+  }
+  var self = this;
+  var keys = Object.keys(cols);
+  if (Array.isArray(fieldsFilter) && fieldsFilter.length > 0) {
+    // No empty array, including all the fields
+    keys = fieldsFilter;
+  } else if ('object' === typeof fieldsFilter &&
+    Object.keys(fieldsFilter).length > 0) {
+    // { field1: boolean, field2: boolean ... }
+    var included = [];
+    var excluded = [];
+    keys.forEach(function(k) {
+      if (fieldsFilter[k]) {
+        included.push(k);
+      } else if ((k in fieldsFilter) && !fieldsFilter[k]) {
+        excluded.push(k);
       }
     });
-  };
+    if (included.length > 0) {
+      keys = included;
+    } else if (excluded.length > 0) {
+      excluded.forEach(function(e) {
+        var index = keys.indexOf(e);
+        keys.splice(index, 1);
+      });
+    }
+  }
+  var names = keys.map(function(c) {
+    return self.columnEscaped(model, c);
+  });
+  return names.join(',');
+};
 
 /**
- * Count all model instances by the where filter
+ * Build a SQL SELECT statement
+ * @param {String} model Model name
+ * @param {Object} filter Filter object
+ * @param {Object} options Options object
+ * @returns {ParameterizedSQL} Statement object {sql: ..., params: [...]}
+ */
+SqlConnector.prototype.buildSelect = function(model, filter, options) {
+  if (!filter.order) {
+    var idNames = this.idNames(model);
+    if (idNames && idNames.length) {
+      filter.order = idNames;
+    }
+  }
+
+  var selectStmt = new ParameterizedSQL('SELECT ' +
+    this.buildColumnNames(model, filter) +
+    ' FROM ' + this.tableEscaped(model)
+  );
+
+  if (filter) {
+
+    if (filter.where) {
+      var whereStmt = this.buildWhere(model, filter.where);
+      selectStmt.merge(whereStmt);
+    }
+
+    if (filter.order) {
+      selectStmt.merge(this.buildOrderBy(model, filter.order));
+    }
+
+    if (filter.limit || filter.skip || filter.offset) {
+      selectStmt = this.applyPagination(
+        model, selectStmt, filter);
```

Comment:
FWIW, I find it personally rather unexpected that an `apply` method is returning a value. I would expect that an `apply` method modifies the data passed in the arguments (i.e. the `selectStmt` object).